pkg/controller/elasticsearch/validation/autoscaling_validation.go

```diff
@@ -0,0 +1,281 @@
+)
+
+const UnexpectedVolumeClaimError = "autoscaling supports only one volume claim"
+
+var (
+	ElasticsearchMinAutoscalingVersion = version.From(7, 11, 0)
+
+	// minMemory is the minimal amount of memory which can be set as the minimum limit in an autoscaling specification.
+	minMemory = resource.MustParse("2G")
+
+	// No minimum values are expected for CPU and Storage.
+	// If provided the validation function must ensure that the value is strictly greater than 0.
+	minCPU     = resource.MustParse("0")
+	minStorage = resource.MustParse("0")
+)
+
+func validAutoscalingConfiguration(es esv1.Elasticsearch) field.ErrorList {
+	if !es.IsAutoscalingDefined() {
+		return nil
+	}
+	proposedVer, err := version.Parse(es.Spec.Version)
+	if err != nil {
+		return field.ErrorList{
+			field.Invalid(field.NewPath("spec").Child("version"), es.Spec.Version, parseVersionErrMsg),
+		}
+	}
+
+	var errs field.ErrorList
+	if !proposedVer.IsSameOrAfter(ElasticsearchMinAutoscalingVersion) {
+		errs = append(
+			errs,
+			field.Invalid(
+				field.NewPath("metadata").Child("annotations", esv1.ElasticsearchAutoscalingSpecAnnotationName),
+				es.Spec.Version,
+				autoscalingVersionMsg,
+			),
+		)
+		return errs
+	}
+
+	// Attempt to unmarshall the proposed autoscaling spec.
+	autoscalingSpecification, err := es.GetAutoscalingSpecification()
+	if err != nil {
+		errs = append(errs, field.Invalid(
+			field.NewPath("metadata").Child("annotations", esv1.ElasticsearchAutoscalingSpecAnnotationName),
+			es.AutoscalingSpec(),
+			err.Error(),
+		))
+		return errs
+	}
+
+	// Validate the autoscaling policies
+	errs = append(errs, validAutoscalingPolicies(autoscalingSpecification.AutoscalingPolicySpecs)...)
+
+	// Get the list of NodeSets managed by an autoscaling policy. This requires to parse the `node.roles` field in the
+	// node configuration, which may raise an error.
+	autoscaledNodeSets, nodeSetConfigErr := autoscalingSpecification.GetAutoscaledNodeSets()
+	if nodeSetConfigErr != nil {
+		errs = append(
+			errs,
+			field.Invalid(
+				field.NewPath("spec").Child("nodeSets").Index(nodeSetConfigErr.Index).Child("config"),
+				nodeSetConfigErr.NodeSet.Config,
+				fmt.Sprintf("cannot parse nodeSet configuration: %s", nodeSetConfigErr.Error()),
+			),
+		)
+		// We stop the validation here as the named tiers are required to validate further.
+		return errs
+	}
+
+	// We want to ensure that an autoscaling policy is at least managing one nodeSet.
+	policiesWithNodeSets := autoscaledNodeSets.AutoscalingPolicies()
+	for i, policy := range autoscalingSpecification.AutoscalingPolicySpecs {
+		if !policiesWithNodeSets.Has(policy.Name) {
+			// No nodeSet matches this autoscaling policy
+			errs = append(
+				errs,
+				field.Invalid(autoscalingSpecPath(i, "roles"),
+					policy.Roles,
+					"roles must be set in at least one nodeSet"),
+			)
+		}
+	}
+
+	// Only one volume claim is supported when a NodeSet is managed by the autoscaling controller.
```

Comment:
First version of the data deciders, especially the proactive one, do not support multiple data paths. Also it would require to discuss how storage is scaled vertically by the operator in that case. I'll add a comment.